### PR TITLE
Add support for IJointFault interface to visualize why a joint is in HW fault on the Yarpmotorgui

### PR DESCRIPTION
**Pull request by mfussi66**
This PR aims to update Yarp 3.5 with cherry-picked commits from `master` regarding the creation of a new interface called IJointFault. The interface is used to communicate the last hardware fault message occurred on a joint to a client, i.e. the Yarpmotorgui.
The feature comes thanks to @randaz81, after an agreement on porting it to 3.5 .

### CODE REVIEW

**Comment by pattacini** [comment]
cc @traversaro

**Review by traversaro — APPROVED** [review]

File: doc/release/master/yarpmotogui_hwfault.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+### GUIs
+
+#### yarpmotorgui
+
+* Modified yarpmotorgui to support a new page "Hardware Fault" which displays the internal hardware error
+
+
+
+
```
Comment (traversaro) [inline doc/release/master/yarpmotogui_hwfault.md:35]: I guess this needs to be moved to another directory?

Comment (mfussi66) [reply doc/release/master/yarpmotogui_hwfault.md]: Moved to the related release folder. Or maybe it's better to update the release notes document?

**Comment by mfussi66** [comment]
> Is it obtained by just a cherry-pick or you did also manual edits?

Just 3 cherry-picks, the only manual edit is in the latest commit.

**Comment by pattacini** [comment]
Hi @randaz81 @mfussi66 @S-Dafarra @traversaro 
After a few discussions happened in the background, we agreed on sticking to the upcoming yarp-3.6 release to have this new functionality. Hence, there's no need for yarp-3.5.2 and this cherry-pick.

Closing.

**Pull request closed without merging**